sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ImageEmbeddingsAsyncClient.java

```diff
@@ -0,0 +1,236 @@
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * Return the embedding vectors for given images.
+     * The method makes a REST API call to the `/images/embeddings` route on the given endpoint.
+     * <p><strong>Header Parameters</strong></p>
+     * <table border="1">
+     * <caption>Header Parameters</caption>
+     * <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>
+     * <tr><td>extra-parameters</td><td>String</td><td>No</td><td>Controls what happens if extra parameters, undefined
+     * by the REST API,
+     * are passed in the JSON request payload.
+     * This sets the HTTP request header `extra-parameters`. Allowed values: "error", "drop", "pass-through".</td></tr>
+     * </table>
+     * You can add these to a request with {@link RequestOptions#addHeader}
+     * <p><strong>Request Body Schema</strong></p>
+     * 
+     * <pre>{@code
+     * {
+     *     input (Required): [
+     *          (Required){
+     *             image: String (Required)
+     *             text: String (Optional)
+     *         }
+     *     ]
+     *     dimensions: Integer (Optional)
+     *     encoding_format: String(base64/binary/float/int8/ubinary/uint8) (Optional)
+     *     input_type: String(text/query/document) (Optional)
+     *     model: String (Optional)
+     *      (Optional): {
+     *         String: Object (Required)
+     *     }
+     * }
+     * }</pre>
+     * 
+     * <p><strong>Response Body Schema</strong></p>
+     * 
+     * <pre>{@code
+     * {
+     *     data (Required): [
+     *          (Required){
+     *             embedding: BinaryData (Required)
+     *             index: int (Required)
+     *         }
+     *     ]
+     *     usage (Required): {
+     *         prompt_tokens: int (Required)
+     *         total_tokens: int (Required)
+     *     }
+     *     model: String (Required)
+     * }
+     * }</pre>
+     *
+     * @param embedRequest1 The embedRequest1 parameter.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return representation of the response data from an embeddings request.
+     * Embeddings measure the relatedness of text strings and are commonly used for search, clustering,
+     * recommendations, and other similar scenarios along with {@link Response} on successful completion of
+     * {@link Mono}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    Mono<Response<BinaryData>> embedWithResponse(BinaryData embedRequest1, RequestOptions requestOptions) {
```

Comment:
The name `embedRequest1` is a bit odd. Why do we have `1` in the param name? Is this coming from typespec or is this generated by the Java emitter?

Reply:
see below

Reply:
So in the TypeSpec both the EmbeddingsClient and ImageEmbeddingsClient have an embed operation, meaning they both generate an implementation "embedRequest" class. I've created a workaround customization to rename the class, and I'll file an issue with the typespec-java DPG team asking about this corner case to see if there's some way to rename that emitted implementation class using TypeSpec.